drivers/wifi/nrf700x/osal/os_if/src/osal.c

```diff
@@ -756,3 +756,11 @@ int wifi_nrf_osal_bus_qspi_ps_status(struct wifi_nrf_osal_priv *opriv,
 	return opriv->ops->bus_qspi_ps_status(os_qspi_priv);
 }
 #endif /* CONFIG_NRF_WIFI_LOW_POWER */
+
+void wifi_nrf_osal_assert(struct wifi_nrf_osal_priv *opriv,
```

Comment:
Why dont' we use a generic macro directly defined by the shim? This is is unnecessary and also isn't giving flexibility is using other conditions than `!=`. 
E.g.,

In Zephyr:

#define WIFI_NRF_OSAL_ASSERT ASSERT()

In Linux:

#define WIFI_NRF_OSAL_ASSERT BUG_ON()

Reply:
Well, this won't work directly as we can't include SHIM headers in OSAL code.